src/rgw/rgw_putobj_processor.cc

```diff
@@ -331,6 +331,10 @@ int AtomicObjectProcessor::complete(size_t accounted_size,
 
   r = obj_op.write_meta(dpp, actual_size, accounted_size, attrs, y);
   if (r < 0) {
+    if (r == -ETIMEDOUT) {
+      // head obj may eventually be written successfully due to network jitter. In such scenario, must keep the set of objects avoiding data corruption.
+      writer.clear_written();
+    }
```

Comment:
thanks, this comment is really helpful. can you please wrap it under ~80 characters, and add a bit more about the other case?
```c++
// if the head object write doesn't succeed, we'll orphan any tail objects as if we'd crashed before that write
```

Reply:
already updated